src/networking.c

```diff
@@ -2758,18 +2758,23 @@ NULL
     }
 }
 
-/* HELLO <protocol-version> [AUTH <user> <password>] [SETNAME <name>] */
+/* HELLO [<protocol-version> [AUTH <user> <password>] [SETNAME <name>] ] */
 void helloCommand(client *c) {
-    long long ver;
+    long long ver = 0;
+    int next_arg = 1;
 
-    if (getLongLongFromObject(c->argv[1],&ver) != C_OK ||
-        ver < 2 || ver > 3)
-    {
+    if (c->argc >= 2 && getLongLongFromObject(c->argv[next_arg++],&ver) == C_OK &&
+            (ver < 2 || ver > 3)) {
         addReplyError(c,"-NOPROTO unsupported protocol version");
         return;
     }
```

Comment:
i think it should look like this:
```c
    if (c->argc >= 2) {
        if (getLongLongFromObjectOrReply(c->argv[next_arg++],&ver, NULL) != C_OK)
            return;
        if (ver < 2 || ver > 3) {
            addReplyError(c,"-NOPROTO unsupported protocol version");
            return;
        }
    }
````

or if want:
```c
    if (c->argc >= 2 && getRangeLongFromObjectOrReply(c->argv[next_arg++],2,3,&ver, "-NOPROTO unsupported protocol version") != C_OK)
        return;
```

instead of the later check you added.
i.e. the problem in the current code of the PR is that `getLongLongFromObject` may fail, and we don't `return`.

Reply:
Yeah, much clear. Thx :), I am a bit lost about the context of this PR.